### PR TITLE
Refactor application status card

### PR DESCRIPTION
**Pull request by vinit717**
Date: `10-01-2024`

Developer Name: `vinit`

---

## Issue Ticket Number:-

- #832 

## Description:

Refactor the status card to show icons for different statuses

Is Under Feature Flag

- [x] Yes
- [ ] No

Database changes

- [ ] Yes
- [x] No

Breaking changes (If your feature is breaking/missing something please mention pending tickets)

- [ ] Yes
- [x] No

Is Development Tested?

- [x] Yes
- [ ] No

Tested in staging?

- [x] Yes
- [ ] No

### Add relevant Screenshot below ( e.g test coverage etc. )
image
image
image

### follow up Issue
- https://github.com/Real-Dev-Squad/website-www/issues/839

### CODE REVIEW

File: tests/integration/components/status-card-test.js
Comment (satyam73) [inline tests/integration/components/status-card-test.js:1]: More tests can be added for other status as well

**Review by satyam73 — CHANGES_REQUESTED** [review]

**Comment by satyam73** [comment]
Can we have rejected icon color to be red?

**Comment by vinit717** [comment]
> Can we have rejected icon color to be red?

Can't get it can you elaborate more?

File: tests/integration/components/status-card-test.js
Comment (vinit717) [reply tests/integration/components/status-card-test.js:1]: added one more case if more tes t should be added then I'll cover in different PR

**Comment by satyam73** [comment]
> > Can we have rejected icon color to be red?
> 
> Can't get it can you elaborate more?

This icon will look good in red imo
image

File: tests/integration/components/status-card-test.js [hunk 1]
```diff
@@ -83,4 +83,46 @@ module('Integration | Component | status-card', function (hooks) {
...
+    `);
+
+    assert.dom('[data-test-status-card-heading]').hasText('Accepted');
+    assert.dom('[data-test-icon="accepted"]').exists();
+    assert
+      .dom('[data-test-status-card-description-1]')
+      .hasText('Congratulations! Your application is accepted by us');
+    assert
+      .dom('[data-test-status-card-description-2]')
+      .doesNotExist("Here's the feedback for your application");
```
Comment (satyam73) [inline tests/integration/components/status-card-test.js:108]:
```suggestion
      .doesNotExist();
```

File: tests/integration/components/status-card-test.js
Comment (satyam73) [reply tests/integration/components/status-card-test.js:1]: Please create and attach ticket here

**Review by satyam73 — CHANGES_REQUESTED** [review]

File: tests/integration/components/status-card-test.js
Comment (vinit717) [reply tests/integration/components/status-card-test.js:1]: https://github.com/Real-Dev-Squad/website-www/issues/839

**Review by prakashchoudhary07 — APPROVED** [review]

**Review by satyam73 — APPROVED** [review]

**Pull request merged**